lib/HTMLParser.js

```diff
@@ -2508,7 +2510,7 @@ function HTMLParser(address, fragmentContext, options) {
 
   // Insert a token, either using the current parser insertion mode
   // (for HTML stuff) or using the insertForeignToken() method.
-  var insertToken = function insertToken(t, value, arg3, arg4) {
+  function insertToken(t, value, arg3, arg4) {
```

Comment:
Couldn't we write this as:
```
var insertToken = HTMLParser.insertToken = function insertToken(....) { ... }
```
instead of using magic function hoisting?  The linter would be much happier in that case as well.